### PR TITLE
nvme - Munin plugin to monitor the use of NVMe devices

### PR DESCRIPTION
**Pull request by kjetilho**
This is a multigraph plugin which makes three graphs, nvme_usage, nvme_bytes and nvme_writecycles.  See POD below for details.

### CODE REVIEW

File: plugins/disk/nvme [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+use strict;
+use Munin::Plugin;
+
+# Check that multigraph is supported
+need_multigraph();
+
+sub run_nvme {
+    my (@cmd) = @_;
+    my @lines;
+    $ENV{'LC_ALL'} = 'C';
```
[the remark this answers is not in the document]
Comment (kjetilho) [reply plugins/disk/nvme]: yes, you are right, in lib/Munin/Node/Service.pm:prepare_plugin_environment

File: plugins/disk/nvme [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    }
+    return \%info;
+}
+
+use Data::Dumper;
+
+my $mode = ($ARGV[0] or "print");
+
+my $list = nvme_list();
+if ($mode eq 'autoconf') {
```
Comment (sumpfralle) [inline plugins/disk/nvme]: Here you need to check for the existence of the nvme executable, or?

File: plugins/disk/nvme [hunk 3]
```diff
@@ -0,0 +1,244 @@
+#! /usr/bin/perl -w
+# -*- perl -*-
+
+=head1 NAME
+
+nvme - Munin plugin to monitor the use of NVMe devices
```
Comment (sumpfralle) [inline plugins/disk/nvme]: Maybe add a short explanation for `NVMe`?

File: plugins/disk/nvme (hunk 2 above)
Comment (kjetilho) [reply plugins/disk/nvme]: well ..  I think the current failure mode is OK?

```
  # munin-run nvme autoconf
  Can't exec "nvme": No such file or directory at /etc/munin/plugins/disk/nvme line 89.
  no (no devices to monitor)
```
checking lsmod output is a possibility to give a better suggestion to the admin, but that would tie the plugin even closer to Linux.  is autoconf supposed to be silent on stderr?

Comment (niclan) [reply plugins/disk/nvme]: Yes, it's supposed to be silent on stderr.

**Comment by kjetilho** [comment]
okay, I've made it more helpful, I hope.  it will still report on stderr in the case where it's not running as root.
```
  # munin-run --servicedir ~ nvme autoconf
  /dev/nvme1n1: Permission denied
  nvme: probably needs to run as user root
  no (no devices to monitor)
```
(the first two lines are on stderr.)  When run as root it will give the appropriate message without noise on stderr.

**Comment by kjetilho** [comment]
btw, IPC::Cmd has been in Perl's corelist since v5.9.5, so should be ok to add without additional requirements.

**Comment by sumpfralle** [comment]
Sorry for letting this rest for so long ...

Thank you!

**Pull request merged**